src/visiomode/devices/__init__.py

```diff
@@ -12,9 +13,20 @@
 import visiomode.plugins as plugins
 
 
-def get_available_devices():
-    """Return list of all serial devices connected to the machine."""
-    return [dev.device for dev in ports.comports()]
+def get_available_devices() -> list[str]:
+    """
+    Return list of all serial devices connected to the machine, with ones defined in the
+    config prepended.
+    """
+    # Get an ordered set of keys so that options presented to the user are more
+    # pleasingly organised.
+    return list(
+        dict.fromkeys(
+            [conf.Config().reward_device_address]
+            + [conf.Config().input_device_address]
+            + [dev.device for dev in ports.comports()]
+        ).keys()
+    )
```

Comment:
I'm confused by this pattern -- why not return a sorted set instead? (i.e. `sorted(set(....]))`

Reply:
The idea was to have an ordered set as in sorted by insertion order, not alphabetical (or otherwise). Since ordered sets are not a thing but now dictionaries are, I was using the keys to keep items sorted in the order they were added while also removing duplicates (e.g., if the config is a serial port that is detected by `pyserial`).

The reason I wanted an ordered set was that the interface shows a dropdown menu for the reward address all the time (i.e., why it's added first to the keys) but the input device address is only required for some tasks. So, adding the addresses in order: reward -> input -> automatically detected, means that the user hopefully gets the right defaults on the interface.

If that's not clear enough from the code I can fix the comment to make it more explicit, or if you think it's overcomplicating things, I can also just remove it and return a simple sorted set.